apps/sms/services/js/bridge_client_mixin.js

```diff
@@ -0,0 +1,144 @@
+/*global bridge */
+/*jshint esnext: true */
+
+'use strict';
+
+(function(exports) {
+
+  const priv = Object.freeze({
+    client: Symbol('client'),
+    serviceName: Symbol('serviceName'),
+    appInstanceId: Symbol('appInstanceId'),
+    pendingOperations: Symbol('pendingOperations'),
+
+    initBridge: Symbol('initBridge'),
+    registerOperation: Symbol('registerOperation')
+  });
+
+  /**
+   * This mixin allows to expose easily a service from an object.
+   *
+   * This needs the following properties to work properly:
+   *
+   * - SERVICE_NAME holds the service name.
+   */
+  var clientMixin = {
+    /**
+     * Initialize service to listen on the default context and given endpoint.
+     * @param {(Iframe|Worker|MessagePort|BroadcastChannel|Window)} endpoint
+     *  The context/thread that service could listen for.
+     */
+    initClient({ appInstanceId, endpoint, plugins }) {
+      if (!appInstanceId) {
+        throw new Error('Application instance identifier is required!');
+      }
+
+      this[priv.appInstanceId] = appInstanceId;
+      this[priv.initBridge](endpoint || exports, plugins);
+    },
+
+    clientMethod(...params) {
```

Comment:
Those methods are just to track promises for pending operations.

@wilsonpage Is it possible (out of the box) to destroy client/service so that it automatically waits for the pending operations to complete? 

More context, in this PR we want to perform the following actions:
* Run app using service hosted in main window (to not pay SharedWorker initialization cost);
* On "visuallyLoaded" we'd like to spawn the same service in a SharedWorker;
* Starting from this moment all new method/stream calls will be forwarded to service hosted in a SharedWorker;
* Since window service stream is still active (basically until "fullyLoaded") we don't want to interrupt it (and to have less complexity), but rather wait for its completion and only then shutdown old bridge.client and bridge.service.